src/remote_adapter/remote_adapter_client/qt_impl/sdl_remote_adapter_qt_client.cc

```diff
@@ -0,0 +1,90 @@
+#include <iostream>
+
+#include "qt_impl/sdl_remote_adapter_qt_client.h"
+#include "qt_impl/sdl_remote_adapter_receive_thread.h"
+
+namespace lua_lib {
+
+SDLRemoteTestAdapterQtClient::SDLRemoteTestAdapterQtClient(
+            const std::string& host,
+            uint32_t port,
+            MqParams& in_params,
+            MqParams& out_params,
+            QObject* parent)
+        : QObject(parent),
+          host_(host),
+          port_(port),
+          in_mq_params_(in_params),
+          out_mq_params_(out_params)  { }
+
+SDLRemoteTestAdapterQtClient::~SDLRemoteTestAdapterQtClient() {
+    if (nullptr != listener_ptr_) {
+        delete(listener_ptr_);
+    }
+
+    if (nullptr != remote_adapter_client_ptr_) {
+        remote_adapter_client_ptr_->clear();
+        delete(remote_adapter_client_ptr_);
+    }
+}
+
+void SDLRemoteTestAdapterQtClient::connectMq() {
+    if (!isconnected_) {
+        try {
+            remote_adapter_client_ptr_ = new SDLRemoteTestAdapterClient(host_, port_);
+            int in_open_result = openWithParams(in_mq_params_);
+            int out_open_result = openWithParams(out_mq_params_);
+            if ((0 == in_open_result) && (0 == out_open_result)) {
+                listener_ptr_ = new SDLRemoteTestAdapterReceiveThread(this);
```

Comment:
We can use shared or unique pointer
instead of simple pointers ?

Reply:
@SNiukalov `std::unique_ptr` is used for pointers on SDLRemoteTestAdapterClient and SDLRemoteTestAdapterReceiveThread instances 
(see bc3f86577b740bc84201d7d0edaf82cce1d8ec52)